primitives/src/conststr.rs

```diff
@@ -469,139 +470,6 @@ impl TryFrom<String> for InlinedString {
     }
 }
 
-mod ffi {
```

Comment:
when FFI function is generated for `Account::roles` it returns `Vec<&RoleId>` which ends up translated into `Vec<&ConstString>`. In the current implementation `Vec<T>` cannot be returned from an FFI function if T carries ownership

I'm reviving the #2669. It's too bothersome to deal with this now

Reply:
if you take `Domain::asset_definitions` this function when converted to an FFI function will obviously return a `Vec<T>`. On the other hand, making an entire `Vec<T>` opaque would incur an FFI call for every access. I grant that you could use some the predefined queries like `FindAllAssetDefinitions` or similar, which seems inconvenient.

Reply:
We can always change it later. I wonder how much binary size we'd be trading, and if we can leverage this to reduce the size even further.